component/async/kafka/kafka.go

```diff
@@ -118,18 +118,23 @@ func (m *message) Ack() error {
 	return nil
 }
 
-// Source returns the kafka topic where the message arrived.
-func (m *message) Source() string {
-	return m.msg.Topic
-}
-
 // Nack signals the producing side an erroring condition or inconsistency.
 func (m *message) Nack() error {
 	messageConfirmationCountInc("NACK")
 	trace.SpanError(m.span)
 	return nil
 }
 
+// Source returns the kafka topic where the message arrived.
+func (m *message) Source() string {
```

Comment:
I just swapped Source() and Nack() so that we have the same order in all the implem & the interface